### PR TITLE
Disable fees when feenominator is set to zero

### LINKED ISSUE #271: Add some way to disable fees
See Swivel.sol.

### PR DESCRIPTION
**Pull request by sourabhmarathe**
This PR addresses issue #271. 

We now get the fee by calling an `internal view` method called `computeFee`. If `feenominator` is 0, then we return a fee of 0. Otherwise, we compute the fee. 

We also set a `MIN_FEENOMINATOR` which sets a maximum rate for fees. This is in keeping with the pattern set forth in [Swivel.sol](https://github.com/Swivel-Finance/swivel/blob/main/contracts/v2/swivel/Swivel.sol#L419).

### CODE REVIEW

File: build/lender/Lender.sol [hunk 1]
```diff
@@ -41,9 +42,17 @@ contract Lender {
...
   /// @return true if successful
   function setFee(uint256 f) external authorized(admin) returns (bool) {
+    require(f >= MIN_FEENOMINATOR, 'fee too high');
     feenominator = f;
     return true;
   }
+
+  /// @notice Disables fees
+  /// @return true if successful
+  function disableFees() external authorized(admin) returns (bool) {
```
Comment (robrobbins) [inline build/lender/Lender.sol]: nah. this is not needed. 

make `setFee` able to take 0.

this obvs mean the `f >= MIN_FEENOMINATOR` would need to make an exception for 0 but thats far superior that writing another method to do the same thing.

File: build/lender/Lender.sol [hunk 2]
```diff
@@ -389,16 +392,30 @@ contract Lender {
...
 
+  /// @notice This method returns the fee based on the amount passed to it. If
+  /// feenominator is 0, then there is no fee.
+  /// @param a Amount of underlying tokens to calculate the fee for
+  /// @return uint256 The total for for the given amount
+  function calculateFee(uint256 a) internal view returns (uint256) {
+    if (feenominator == 0) {
+      return 0;
+    } 
+    return a / feenominator;
```
Comment (robrobbins) [inline build/lender/Lender.sol]: solidity does have a ternary, use it

`return feenominator > 0 ? a / feenominator : 0`

also can this internal function be `pure` vs `view` ? do it if so.

File: test/lender/Lender.sol [hunk 3]
```diff
@@ -41,9 +42,17 @@ contract Lender {
...
   /// @return true if successful
   function setFee(uint256 f) external authorized(admin) returns (bool) {
+    require(f >= MIN_FEENOMINATOR, 'fee too high');
     feenominator = f;
     return true;
   }
+
+  /// @notice Disables fees
+  /// @return true if successful
+  function disableFees() external authorized(admin) returns (bool) {
```
Comment (robrobbins) [inline test/lender/Lender.sol]: same, use setFee that can take 0

File: test/lender/Lender.sol [hunk 4]
```diff
@@ -402,6 +405,17 @@ contract Lender {
...
 
+  /// @notice This method returns the fee based on the amount passed to it. If
+  /// feenominator is 0, then there is no fee.
+  /// @param a Amount of underlying tokens to calculate the fee for
+  /// @return uint256 The total for for the given amount
+  function calculateFee(uint256 a) internal view returns (uint256) {
+    if (feenominator == 0) {
+      return 0;
+    } 
+    return a / feenominator;
```
Comment (robrobbins) [inline test/lender/Lender.sol]: same..

**Review by robrobbins — DISMISSED** [review]

File: build/lender/Lender.sol (hunk 2 above)
Comment (sourabhmarathe) [reply build/lender/Lender.sol]: It cannot be `pure` due to the reference to `feenominator`

**Review by robrobbins — APPROVED** [review]

**Pull request merged**